gamemodes/terrortown/gamemode/server/sv_shop.lua

```diff
@@ -97,18 +97,7 @@ local function OrderEquipment(ply, cls)
 	-- keep compatibility with old addons
 	if not hook.Run("TTTCanOrderEquipment", ply, idOrCls, is_item) then return end
 
-	-- add our own hook with more consistent class parameter and some more information
-	local allow, ignoreCost, message = hook.Run("TTT2CanOrderEquipment", ply, cls, is_item, credits)
-
-	if message then
-		LANG.Msg(ply, message, nil, MSG_MSTACK_ROLE)
-	end
-
-	if allow == false then return end
-
-	if not ignoreCost then
-		ply:SubtractCredits(credits)
-	end
+	ply:SubtractCredits(credits)
```

Comment:
You missed the `ignoreCost` case, as the player is not charged any credits when ignoreCost is true

Reply:
When I added that hook in #441 that was the intention, but it was problematic in a sense. You could let someone order something and then give it to them for free. You could however not let them order something they couldn't afford in the first place. Maybe EquipmentIsBuyable should just return an additional boolean ignoreCost, that we can use in the OrderEquipment function. That way we would keep the functionality and still open it up to shared realm. What do you think?

Reply:
I'm not deep in this topic, but it sounds good to me

Reply:
I don't really like this approach, as the function is not meant to judge about whether or not the cost is to be ignored. The function is even called EquipmentIsBuyable. So we might have to think about another way to address this, but i get why you want to change it and that should definitely be addressed.

Reply:
All alternative I can think of lead to code redundancy

Reply:
Redundancy is not the ultimate evil, as sometimes redundancy is better than trying to "eliminate" it, by making the code less flexible or even much more unreadable etc... 
We should still think about another way or even rethink the whole function etc.